lib/model.js

```diff
@@ -3939,7 +3939,16 @@ function getModelsMapForPopulate(model, docs, options) {
       foreignField = foreignField.call(doc);
     }
 
-    const ret = convertTo_id(utils.getValue(localField, doc));
+    const localFieldPath = modelSchema.paths[localField];
+    const localFieldGetters = localFieldPath ? localFieldPath.getters : [];
+    let ret;
+
+    if (localFieldGetters.length) {
+      ret = localFieldGetters[0].call(doc, doc[localField]);
```

Comment:
Why do this instead of `localFieldPath.applyGetters(doc[localField], doc)` ?

Reply:
D'oh. I should have been able to find that. For some strange reason I looked at the api docs for a solution instead of grep'ing the source code like I usually do. fixed.